MV.Wildebeest.Core/source/core/java/co/mv/wb/impl/WildebeestApiImpl.java

```diff
@@ -788,23 +788,36 @@ private static void validateMigrationStates(
 				{
 					migrationToStateValid = true;
 				}
-				if (m.getToState().equals(s.getStateId()) || m.getToState().equals(s.getLabel()))
+				if (m.getFromState().equals(s.getStateId()) || m.getFromState().equals(s.getLabel()))
 				{
 					migrationFromStateValid = true;
 				}
 
-				if (migrationFromStateValid == true && migrationToStateValid == true)
+				if (migrationFromStateValid && migrationToStateValid)
 				{
 					break;
 				}
 			}
 
-			if (migrationFromStateValid == false || migrationToStateValid == false)
+			if (!migrationFromStateValid)
+			{
+				throw new InvalidReferenceException
+					(
+						STATE,
+						m.getFromState().get(),
+						m.getApplicableTypes().toString(),
```

Comment:
The referring type in this case is "migration" - this can be another constant like "STATE".